### ISSUE TITLE
Annotations on machine inventory from /etc/os-release are not updated

### ISSUE THREAD

**Issue by davidcassany** [rancher/elemental-operator#811]
After upgrading nodes from latest stable release (v2.1.1) to staging OS (v2.1.2) the annotation  `elemental.cattle.io/os.image` is not updated to the new image reference. 

Elemental-operator version: v1.6.3 (staging build as of today)

**Comment by anmazzotti** [comment]
Please consider reworking or adding to this documentation: https://elemental.docs.rancher.com/upgrade-lifecycle#verifying-successful-upgrades

**Comment by fgiudici** [comment]
> Please consider reworking or adding to this documentation: https://elemental.docs.rancher.com/upgrade-lifecycle#verifying-successful-upgrades

Makes sense, tracked in a separated issue:
https://github.com/rancher/elemental-docs/issues/360

**Comment by fgiudici** [comment]
The OS annotations are not updated because the `elemental-registration.service` is never triggered.
This is due to the `elemental-registration.timer` being disabled. Re-enabling the timer, will call the elemental-registry binary for registration at each boot.
Fixing this anyway will not be enough: the elemental-register binary will refuse to run a plain re-registration before 24 hours, so an OS update may be reflected int he annotations only after 24 hours from the last registration labels/annotations refresh.
So, to recap, two fixes are needed:
1. enable elemental-register.timer unit in the OS image
2. rework the re-registration behavior to allow refresh of annotations (and labels) at each boot
